service/docs/source/geopm_pio_sst.7.rst

```diff
@@ -0,0 +1,197 @@
+-----------
+
+The SSTIOGroup implements the `geopm::IOGroup(3)
+<GEOPM_CXX_MAN_IOGroup.3.html>`_ interface to provide hardware signals
+and controls for key Intel Speed Select Technology features on
+supported Intel platforms with Linux Kernel versions >=5.3
+
+
+SST-CP: Core Power
+~~~~~~~~~~~~~~~~~~~
+
+SST Core Power feature enables the user to specify priority of cores
+for power distribution in a power-constrained scenario. Cores are
+assigned to different Class Level of Service (CLOS). Each CLOS has a
+defined min/max frequency and weight, which determines how the
+frequency is allocated.
+
+
+SST-TF: Turbo Frequency
+~~~~~~~~~~~~~~~~~~~~~~~
+
+SST Turbo Frequency feature enables the user to dynamically set
+different turbo ratio limits for different cores. Cores are specified
+as high-priority or low-priority via the SST-CP feature. Cores in CLOS
+0 or 1 are high priority and can reach higher than the all-core turbo
+ratio limit when there is sufficient power headroom. Cores in CLOS 2
+or 3 are constrained below the all-core turbo ratio limit. This
+feature only works when SST-CP is enabled.
+
+SIGNALS
+-------
+
+SYSTEM INFO
+~~~~~~~~~~~
+
+* ``SST::CONFIG_LEVEL``:
+  (Package scope) Returns the system's configuration level (SST-PP
+  feature)
+
+* ``SST::COREPRIORITY_SUPPORT``:
+  (Package scope) Returns 1 if SST-CP feature is supported, 0 if
+  unsupported.
+
+* ``SST::TURBOFREQ_SUPPORT``:
+  (Package scope) Returns 1 if SST-TF feature is supported, 0 if
+  unsupported.
+
+* ``SST::HIGHPRIORITY_NCORES:n``:
+  (Package scope) Returns the count of high priority turbo frequency
+  cores in bucket n. Buckets 0 - 7 are supported. Buckets are defined
+  by the number of high priority cores (cores in CLOS 0 or 1), and
+  they determine the frequencies that are obtainable by those
+  cores. Generally, if there are fewer high priority cores, the
+  increase in turbo frequency limit is greater.
+
+* ``SST::HIGHPRIORITY_FREQUENCY_SSE:n``:
+  (Package scope) Returns the high priority turbo frequency for bucket
+  n at the SSE license level.
+
+* ``SST::HIGHPRIORITY_FREQUENCY_AVX2:n``:
+  (Package scope) Returns the high priority turbo frequency for bucket
+  n at the AVX2 license level.
+
+* ``SST::HIGHPRIORITY_FREQUENCY_AVX512:n``:
+  (Package scope) Returns the high priority turbo frequency for bucket
+  n at the AVX512 license level.
+
+* ``SST::LOWPRIORITY_FREQUENCY:[SSE|AVX2|AVX512]``:
+  (Package scope) RReturns the low-priority turbo frequency of the
+  specified licence level. Note these frequencies do not change based
+  on the number of high priority cores.
+
+
+CONFIGURATION
+~~~~~~~~~~~~~
+
+* ``SST::TURBO_ENABLE``:
+  (Package scope) Returns 1 if SST-TF feature is enabled, 0 if
+  disabled.
+
+* ``SST::COREPRIORITY_ENABLE``:
+  (Package scope) Returns 1 if SST-CP feature is enabled, 0 if
+  disabled.
+
+* ``SST::COREPRIORITY:ASSOCIATION``:
+  (Core scope) Returns the core's assigned CLOS.
+
+* ``SST::COREPRIORITY:n:WEIGHT``:
```

Comment:
#2332 went in today to rename this to `SST::COREPRIORITY:n:PRIORITY` and scale it to the `[0,1]` range. The ordering is preserved in that change (so 0 is still assigned to the bucket that should get more budget).